runhouse/resources/hardware/cluster.py

```diff
@@ -642,18 +642,13 @@ def restart_server(
                     f"cluster using domain: {domain}."
                 )
             else:
-                # If no cert and keyfile already exist, generate them client side and copy them onto the cluster
-                if (
-                    not Path(self.cert_config.cert_path).exists()
-                    or not Path(self.cert_config.key_path).exists()
-                ):
-                    self.cert_config.generate_certs(address=self.address)
-
+                # Need to include the IP address in the cert's SANs, so rebuild on restart to ensure the
+                # correct IP is included. Cert subject name needs to match the target host name (in this case IP)
+                self.cert_config.generate_certs(address=self.address)
```

Comment:
Not sure I follow, why do we need this?

Reply:
the subject name on the cert needs to match the target, otherwise verifcation will fail. So if we have a domain we can re-use the same cert across multiple servers, but if we don't the IP address will need to be updated. Do you see an alternative?

Reply:
Not sure I follow - with the flow that's in there now I was able to verify a cluster with an existing cert in my filesystem, no customization for IP. What was the error you saw?